src/services/contracts/ContractsInkService.spec.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { ContractPromise } from '@polkadot/api-contract';
+
+import { sanitizeNumbers } from '../../sanitize/sanitizeNumbers';
+import { polkadotRegistryV9190 } from '../../test-helpers/registries';
+import { defaultMockApi } from '../test-helpers/mock';
+import { ContractsInkService } from '.';
+
+const contractInkService = new ContractsInkService(defaultMockApi);
+
+const getFlipper = () =>
+	Promise.resolve().then(() => {
+		return {
+			debugMessage: polkadotRegistryV9190.createType('Text', ''),
+			gasConsumed: polkadotRegistryV9190.createType('u64', '7437907045'),
+			gasRequired: polkadotRegistryV9190.createType('u64', '74999922688'),
+			output: true,
+			result: polkadotRegistryV9190.createType('ContractExecResultResult', {
+				ok: {
+					flags: [],
+					data: '0x01',
+				},
+			}),
+			storageDeposit: polkadotRegistryV9190.createType('StorageDeposit', {
+				charge: '0',
+			}),
+		};
+	});
+
+const mockContractPromise = {
+	query: {
+		get: getFlipper,
+	},
+} as unknown as ContractPromise;
+
+describe('ContractsInkService', () => {
```

Comment:
This doesn't really test that much (the service hardly does anything really, and the rest is mocked out). Makes sense to have; maybe an e2e test against a real life contract would be more reassuring if possible?

Reply:
Yea I agree, I just need to make sure that there is some live contracts on Kusama or westend to test against. But yea I totally agree, an e2e test would be great.

Reply:
So the only issue is, AFAIK there is no live contracts on Westend, Kusama, or Polkadot. Apparently there is some on Rococo, but I am hesitant to add another chain for e2e-tests as it's more to maintain. Let me know what your thoughts are.

Reply:
Ok yeah; I thought it might be a bit tricky!

In the ideal case I guess we'd spin up a test node, deploy a contract to it and then attempt to use this endpoint to query some methods on the contract. Do we have any test infrastructure to help do that already anywhere? Is it in the realms of possibility to do?

Reply:
We actually dont have much infratructure for this currently. IMO It would actually probably just be better to add roccoco to the e2e tests than go through implementing that infrastructure.